dapps/satoshi-multiplayer/contract/sources/tournament.move

```diff
@@ -194,42 +218,110 @@ module contract::tournament {
                 hash_1: _,
                 secret_0,
                 secret_1,
-                round,
+                round: _,
             } = match;
 
-            // Check if player_0 or player_1 has secret.
-            if(option::is_none(&secret_0)) {
-                // If _0 has no secret, make _1 the winner.
+            // Make sure secrets are revealed before deciding the winner
+            assert!((option::is_some(&secret_0) && option::is_some(&secret_1)), ESecretsNotRevealed);
```

Comment:
We don't want to fail here in a case a player hasn't revealed their secret. We want to proceed with player_1 in case player_0 hasn't revealed and vice versa. @Eis-D-Z ?

Reply:
End round should check if the allotted time for the round has passed or not. If the round is going to last for 1 epoch after that time passes it should check if both hashes are present, if not award a win to the one who committed a hash. Then check if both secrets have been committed, if not award a win to the player who did commit. If everything is in order then derive the winner based on the secrets.